### PR TITLE
Drop IP spoofing for outbound traffic

### PR DESCRIPTION
**Pull request by howardjohn**
This is not needed and breaks dualstack + adds complexity

### CODE REVIEW

File: src/proxy/outbound.rs [hunk 1]
```diff
@@ -79,7 +79,8 @@ impl Outbound {
         let (sub_drain_signal, sub_drain) = drain::channel();
         let pool = proxy::pool::WorkloadHBONEPool::new(
             self.pi.cfg.clone(),
-            self.enable_orig_src,
+            // Do not need to spoof with inpod mode for outbound
+            self.enable_orig_src && !self.pi.cfg.inpod_enabled,
```
Comment (bleggett) [inline src/proxy/outbound.rs:83]: Can we just do this check in `new` and set `enable_orig_src = false` once-for-all?

File: src/proxy/socks5.rs [hunk 2]
```diff
@@ -81,7 +81,8 @@ impl Socks5 {
                 // but ProxyInfo is overloaded and only `outbound` should ever use the pool.
                 let pool = crate::proxy::pool::WorkloadHBONEPool::new(
                     self.pi.cfg.clone(),
-                    self.enable_orig_src,
+                    // Do not need to spoof with inpod mode for outbound
+                    self.enable_orig_src && !self.pi.cfg.inpod_enabled,
```
Comment (bleggett) [inline src/proxy/socks5.rs:85]: Ditto - hoist the config check to `new` and just turn it off at the beginning.

File: src/proxy/outbound.rs (hunk 1 above)
Comment (howardjohn) [reply src/proxy/outbound.rs]: good call, done

File: src/proxy/socks5.rs (hunk 2 above)
Comment (howardjohn) [reply src/proxy/socks5.rs]: :+1:

**Review by bleggett — APPROVED** [review]

**Pull request merged**